sql/2019/10_SEO/10_15.sql

```diff
@@ -0,0 +1,11 @@
+#standardSQL
+
+# CruX speed metrics (FCP, server response time)
```

Comment:
This is a great start. I'd combine FCP and FID using the [PageSpeed Insights formula](https://developers.google.com/speed/docs/insights/v5/about#categories) for determining whether a website is fast/avg/slow.

```sql
#standardSQL
# % of websites classified as fast/avg/slow
SELECT
  ROUND(COUNTIF(fast_fcp >= .9 AND fast_fid >= .95) * 100 / COUNT(0), 2) AS pct_fast,
  ROUND(COUNTIF(NOT(slow_fcp >= .1 OR slow_fid >= 0.05) AND NOT(fast_fcp >= .9 AND fast_fid >= .95)) * 100 / COUNT(0), 2) AS pct_avg,
  ROUND(COUNTIF(slow_fcp >= .1 OR slow_fid >= 0.05) * 100 / COUNT(0), 2) AS pct_slow
FROM
  `chrome-ux-report.materialized.metrics_summary`
WHERE
  date = '2019-07-01'
```

And alternatively, here's a breakdown by users' device types:

```sql
#standardSQL
# % of websites classified as fast/avg/slow by form factor
SELECT
  device,
  ROUND(COUNTIF(fast_fcp >= .9 AND fast_fid >= .95) * 100 / COUNT(0), 2) AS pct_fast,
  ROUND(COUNTIF(NOT(slow_fcp >= .1 OR slow_fid >= 0.05) AND NOT(fast_fcp >= .9 AND fast_fid >= .95)) * 100 / COUNT(0), 2) AS pct_avg,
  ROUND(COUNTIF(slow_fcp >= .1 OR slow_fid >= 0.05) * 100 / COUNT(0), 2) AS pct_slow
FROM
  `chrome-ux-report.materialized.device_summary`
WHERE
  yyyymm = '201907'
GROUP BY
  device
```

Reply:
One other approach. Since Google has only announced that speed is a ranking factor for mobile, here's a query that is limited only to phone form factors:

```sql
#standardSQL
# % of websites classified as fast/avg/slow, according to PSI
SELECT
  ROUND(COUNTIF(fast_fcp >= .9 AND fast_fid >= .95) * 100 / COUNT(0), 2) AS pct_fast,
  ROUND(COUNTIF(NOT(slow_fcp >= .1 OR slow_fid >= 0.05) AND NOT(fast_fcp >= .9 AND fast_fid >= .95)) * 100 / COUNT(0), 2) AS pct_avg,
  ROUND(COUNTIF(slow_fcp >= .1 OR slow_fid >= 0.05) * 100 / COUNT(0), 2) AS pct_slow
FROM
  `chrome-ux-report.materialized.device_summary`
WHERE
  yyyymm = '201907' AND
  device = 'phone'
```

That may be more relevant to the SEO chapter.